app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -29,6 +31,37 @@ def send_code
 
     private
 
+    def lock_out_max_2fa_sends
+      return unless exceeded_otp_send_limit?
+
+      UpdateUser.new(
+        user: current_user,
+        attributes: {
+          second_factor_locked_at: Time.zone.now,
+          otp_first_sent_at: nil,
+          otp_send_count: 0,
+        }
+      ).call
+
+      handle_second_factor_locked_user(:otp)
```

Comment:
I'm not sure we want to treat this as a `MULTI_FACTOR_AUTH_MAX_ATTEMPTS` event. I think we need a separate event. Also, we were discussing not locking the user out, and displaying a message such as "Didn't receive your OTP? Please wait a minute before requesting a new one."